src/stream.ts

```diff
@@ -236,7 +245,9 @@ export class RemoteStream extends Stream {
           log.debug('on track called');
           // once media for a remote track arrives, show it in the remote video element
           track.onunmute = () => {
-            resolve(streams[0]);
+            if(streams.length > 0) {
+              resolve(streams[0]);
+            }
```

Comment:
We need to reject here otherwise the promise will be stuck in a pending state

Reply:
I think that was my original issue when both streams were being offered but sfu only sent one. Accessing the empty array was throwing and cause the stream construction to get passed undefined. Should we catch during that process, or leave it up to the users to catch/handle?

Reply:
I guess we should handle it internally if it is an expected case. There isn't really anything the user can do right? They should just noop the rejection anyways?

Reply:
Tested this again with the changes in this branch and its no longer expected to happen. So probably fine to just throw.